src/oscar/apps/dashboard/ranges/views.py

```diff
@@ -144,27 +166,73 @@ def add_products(self, request):
 
         self.handle_query_products(request, range, form)
         self.handle_file_products(request, range, form)
-        return HttpResponseRedirect(self.get_success_url(request))
+        return redirect(
+            reverse('dashboard:range-products', kwargs={'pk': range.pk})
+        )
+
+    def add_excluded_products(self, request):
+        range = self.get_range()
+        form = self.form_class(
+            range, request.POST, request.FILES, initial={
+                "upload_type": RangeProductFileUpload.EXCLUDED_PRODUCTS_TYPE}
+        )
+        if not form.is_valid():
+            ctx = self.get_context_data(form_excluded=form,
+                                        object_list=self.object_list)
+            return self.render_to_response(ctx)
+
+        self.handle_query_products(request, range, form)
+        self.handle_file_products(request, range, form)
+        return redirect(
+            reverse('dashboard:range-products', kwargs={'pk': range.pk})
+            + '?upload_type=excluded'
```

Comment:
I don't think this is really necessary. Once the upload was successful, we can just redirect them as is, maybe with a success message stating it was successful. The GET parameter is not really needed in my opinion.

Reply:
It's used to redirected to the page having the second tab with excluded products active.